lib/src/BouncyCheckbox.js

```diff
@@ -1,130 +1,153 @@
-import styles, { _iconContainer, _textStyle } from "./BouncyCheckbox.style";
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { Animated, Easing, Text, TouchableOpacity, View, StyleSheet } from 'react-native'
+import Icon from 'react-native-dynamic-vector-icons'
 
 class BouncyCheckbox extends Component {
   constructor(props) {
-    super(props);
+    super(props)
     this.state = {
       checked: false,
-      springValue: new Animated.Value(1),
-    };
+      springValue: new Animated.Value(1)
+    }
   }
 
   componentDidMount() {
-    this.setState({ checked: this.props.isChecked });
+    this.setState({ checked: this.props.isChecked })
   }
 
   spring = () => {
-    const { useNativeDriver } = this.props;
-    const { checked, springValue } = this.state;
+    const { useNativeDriver } = this.props
+    const { checked, springValue } = this.state
     this.setState({ checked: !checked }, () => {
-      springValue.setValue(0.7);
+      springValue.setValue(0.7)
       Animated.spring(springValue, {
         toValue: 1,
         friction: 3,
-        useNativeDriver,
-      }).start();
+        useNativeDriver
+      }).start()
       // Outside of the onPress function
-      const { onPress } = this.props;
-      if (onPress) onPress(this.state.checked);
-    });
-  };
+      const { onPress } = this.props
+      if (onPress) {
+        onPress(this.state.checked)
+      }
+    })
+  }
 
   renderCheckIcon = () => {
-    const { checked, springValue } = this.state;
+    const { checked, springValue } = this.state
     const {
+      size,
+      fillColor,
+      unfillColor,
+      borderWidth,
+      borderRadius,
+      borderColor,
       iconName,
       iconSize,
       iconType,
       iconColor,
-      fillColor,
       iconStyle,
-      unfillColor,
-      iconComponent,
-    } = this.props;
+      iconComponent
+    } = this.props
     return (
-      <Animated.View
-        style={[ { transform: [{ scale: springValue }] },
-          _iconContainer(checked, fillColor, unfillColor),  iconStyle,
-        ]}
-      >
-        {iconComponent || (
-          <Icon
-            size={iconSize}
-            name={iconName}
-            type={iconType}
-            color={iconColor}
-          />
-        )}
-      </Animated.View>
-    );
-  };
+        <Animated.View
+            style={[
+              { transform: [{ scale: springValue }] },
+              {
+                width: size,
+                height: size,
+                borderWidth,
+                borderRadius,
+                alignItems: 'center',
+                justifyContent: 'center',
+                borderColor,
+                backgroundColor: checked ? fillColor : unfillColor
+              },
+              iconStyle
+            ]}
+        >
+          {iconComponent || <Icon size={iconSize} name={iconName} type={iconType} color={iconColor} />}
+        </Animated.View>
+    )
```

Comment:
5. from your comment. 
Actually I don't believe that inline styling is a good practice. It is way too much code to read and decrease the code readability. That's why separated styling is a better choice for me, it is even more controllable.

Reply:
Got you will not use inline styling for your projects